machine/v1alpha1/types_openstack.go

```diff
@@ -100,6 +101,28 @@ type OpenstackProviderSpec struct {
 
 	// The subnet that a set of machines will get ingress/egress traffic from
 	PrimarySubnet string `json:"primarySubnet,omitempty"`
+
+	// FailureDomain bundles placement instructions.
+	FailureDomain FailureDomain `json:"failureDomain,omitempty"`
+}
+
+// FailureDomain defines a bundle of fault domains across different modules: compute, storage and network.
```

Comment:
// FailureDomain defines a set of correlated fault domains across different OpenStack services: compute, storage, and network.